### PR TITLE
WIP: Add superclass options to the NewtonsoftCSharpTargetLanguage

### PR DESCRIPTION
**Pull request by bitsol**

### CODE REVIEW

**Comment by bitsol** [comment]
I'm switching branches and fixing build errors now.

File: src/quicktype-core/language/CSharp.ts [hunk 1]
```diff
@@ -507,6 +507,7 @@ export const newtonsoftCSharpOptions = Object.assign({}, cSharpOptions, {
         ["attributes-only", { helpers: false, attributes: true }],
         ["just-types", { helpers: false, attributes: false }]
     ]),
+    superclass: new EnumOption("superclass", "Superclass", [["EntityData", "EntityData"], ["none", undefined]], "none"),
```
Comment (schani) [inline src/quicktype-core/language/CSharp.ts]: Oh, one more thing: Please add the option `"secondary"` at the end.  Then it'll show up in the "Other" tab in the UI.

File: src/quicktype-core/language/CSharp.ts [hunk 2]
```diff
@@ -573,6 +575,9 @@ export class NewtonsoftCSharpRenderer extends CSharpRenderer {
         if (this._options.dense) {
             forbidden.push("J", "R", "N");
         }
+        if (this._options.superclass) {
```
Comment (schani) [inline src/quicktype-core/language/CSharp.ts]: `this._options.superclass !== undefined`

File: src/quicktype-core/language/CSharp.ts [hunk 3]
```diff
@@ -617,8 +622,16 @@ export class NewtonsoftCSharpRenderer extends CSharpRenderer {
             this.emitUsing([denseRequiredEnumName, " = Newtonsoft.Json.Required"]);
             this.emitUsing([denseNullValueHandlingEnumName, " = Newtonsoft.Json.NullValueHandling"]);
         }
+        if (this._options.superclass == "EntityData") {
```
Comment (schani) [inline src/quicktype-core/language/CSharp.ts]: `===`

**Pull request closed without merging**